src/sos/section_analyzer.py

```diff
@@ -356,7 +355,7 @@ def get_step_depends(section):
                     "STEP",
                     f"Args {value} in depends cannot be determined: {e}")
         finally:
-            [env.sos_dict.dict().pop(x) for x in svars]
+            old_values = [env.sos_dict.dict().pop(x) for x in svars]
```

Comment:
Again, `old_values` is useful variable. Please change this expression to a for loop.